pkg/awsutils/awsutils.go

```diff
@@ -698,6 +709,34 @@ func (cache *EC2InstanceMetadataCache) tagENI(eniID string) {
 	})
 }
 
+//parseAdditionalEniTagsMap will create a map for additional tags
+func parseAdditionalEniTagsMap(additionalEniTags string) (map[string]string, error) {
+	var additionalEniTagsMap map[string]string
+
+	// If duplicate keys exist, the value of the key will be the value of latter key.
+	err := json.Unmarshal([]byte(additionalEniTags), &additionalEniTagsMap)
+	if err != nil {
+		log.Errorf("Invalid format for ADDITIONAL_ENI_TAGS. Expected a json hash: %v", err)
+	}
+	return additionalEniTagsMap, err
+}
+
+// MapToTags converts a map to a slice of tags.
+func mapToTags(tagsMap map[string]string) []*ec2.Tag {
+	tags := make([]*ec2.Tag, 0)
```

Comment:
To prevent removing the instance ID/node name tag (see comment above), you need to pass in a slice of tags to append to instead of creating a new slice of tags.

Reply:
You are right, thanks for catching ! added changes as per your comments